typescript/system.ts

```diff
@@ -2,18 +2,20 @@ import { IdResource } from './misc'
 
 export type FormsBuilderAISystemConfiguration = {
   systemPrompt: string
+  topPOrTemperature: 'TOP_P' | 'TEMPERATURE'
+  topP: number
   maxTokensPerResponse: number
-} & ({ topP: number } | { temperature: number }) &
-  (
-    | {
-        shouldReason?: false
-        topK: number
-      }
-    | {
-        shouldReason: true
-        reasoningTokenBudget: number
-      }
-  )
+} & (
+  | {
+      shouldReason?: false
+      topK: number
+      temperature: number
```

Comment:
Wouldn't these be potentially `undefined`?

Reply:
Unless you plan to change the validation

Reply:
Mm, its a bit different with reasoning though, because the temp is used, its just locked at `1` so we show that. (top k should prob be hidden though). Your proposing we show it disabled with an arbitrary value that they set, could be confusing (is it locked at that value? I just can't use it?) idk.

Reply:
I originally voted for not showing things that will not be used but we went with showing things disabled with tooltips saying why it was not being used. Im ok with either approach. These types will support both.